local_peer.go

```diff
@@ -147,10 +158,23 @@ func (peer *localPeer) actorLoop(actionChan <-chan localPeerAction) {
 			action()
 		case <-gossipTimer:
 			peer.router.sendAllGossip()
+		case <-peer.timer.C:
+			peer.broadcastPendingTopologyUpdates()
 		}
 	}
 }
 
+func (peer *localPeer) broadcastPendingTopologyUpdates() {
+	peer.router.Routes.recalculate()
```

Comment:
Because this is async, I think it's unwise to move it here.
I'm not entirely sure it was ever necessary, but left at the original place there is some chance it will happen before the broadcast.

Reply:
> I'm not entirely sure it was ever necessary

I changed it so to avoid recalculate an every add/delete connection. But considering #111 
I reverted to keep original calls to routes.recalculate().